files/initramfs-tools/arista-convertfs

```diff
@@ -0,0 +1,170 @@
+#!/bin/sh
+
+case $1 in
+    prereqs)
+        exit 0
+        ;;
+esac
+
+set -e
+# set -x
+total_mem=$(free | awk '/^Mem:/{print $2}')
+tmpfs_size=$(( $total_mem / 20 *  17 ))
+free_mem_thres=$(( $total_mem / 20 * 18 ))
```

Comment:
The script need a memory-based tmpfs to backup and restore the files in the flash during the conversion. The size of created tmpfs is expected to be as large as possible to backup all files in the flash, but it is constrained by the free memory available.  

In the script, tmpfs_size is the size of tmpfs to create, and we expect that the free memory space is at least free_mem_thres, which is tmpfs_size + some extra.

According to our testing, the script with the current settings can always create a tmpfs, and never fail at the check of free memory space (line 123).  (The script runs as the early stage of booting, so an extremely large portion of the memory should be free.)  A special case is that when tmpfs_size is smaller than the total size of files in the flash (line 136), as there is not enough space to backup all files, the booting will continue without any change to the file system and partition. (But this will not never happen on the boxes with 2G flash and 4G memory.)